saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/assertion/SAMLAttributeValueParser.java

```diff
@@ -141,4 +143,27 @@ public static String parseAnyTypeAsString(XMLEventReader xmlEventReader) throws
         }
     }
 
+    private static String addNamespaceWhenMissing(Deque<String> definedPrefixes, XMLEventWriter writer, StartElement startElement)
+            throws XMLStreamException {
+
+        if (startElement.getName().getPrefix() != null && !startElement.getName().getPrefix().isEmpty()
+                && !definedPrefixes.contains(startElement.getName().getPrefix())) {
+
+            Iterator<Namespace> namespaces = startElement.getNamespaces();
+            boolean hasNamespace = false;
+            while (namespaces.hasNext()) {
+                Namespace namespace = namespaces.next();
+                if (namespace.getNamespaceURI().equals(startElement.getName().getNamespaceURI())) {
+                    hasNamespace = true;
+                    break;
+                }
+            }
+            if (!hasNamespace) {
+                XMLEventFactory xmlEventFactory = XMLEventFactory.newInstance();
```

Comment:
This is a costly operation. Please introduce a method `getXMLEventFactory` similar to `getXMLInputFactory` (below for reference) and use it instead. 
https://github.com/keycloak/keycloak/blob/9b43e62f2b6279ba0da9cedcf4b790ccd9b5edaa/saml-core/src/main/java/org/keycloak/saml/common/util/StaxParserUtil.java#L905-L933